packages/hash/api/src/graphql/resolvers/ontology/data-type.ts

```diff
@@ -0,0 +1,57 @@
+import { ApolloError } from "apollo-server-express";
+import { AxiosError } from "axios";
+
+import {
+  IdentifiedDataType,
+  QueryGetDataTypeArgs,
+  Resolver,
+} from "../../apiTypes.gen";
+import { GraphQLContext } from "../../context";
+import { DataTypeModel } from "../../../model";
+import { NIL_UUID } from "../../../model/util";
+
+export const getAllLatestDataTypes: Resolver<
+  Promise<IdentifiedDataType[]>,
+  {},
+  GraphQLContext,
+  {}
+> = async (_, __, { dataSources }) => {
+  const { graphApi } = dataSources;
+
+  const allLatestDataTypeModels = await DataTypeModel.getAllLatest(graphApi, {
+    /** @todo Replace with User from the request */
+    accountId: NIL_UUID,
+  }).catch((err: AxiosError) => {
+    throw new ApolloError(`${err.response?.data}`, "GET_ALL_ERROR");
+  });
+
+  return allLatestDataTypeModels.map(
+    (dataType) =>
+      <IdentifiedDataType>{
+        createdBy: dataType.accountId,
+        dataTypeVersionedUri: dataType.schema.$id,
+        schema: dataType.schema,
+      },
+  );
```

Comment:
I think we should encapsulate the mapping to a GQL type in a `toGQL` method so that this logic becomes reusable. Previously we made this a method on the model classes.

Reply:
I think this does make a lot of sense, but I did dislike combining layers in the model classes (e.g. having model classes import from the database files and the GQL generated files). I don't mind adding it, but I just want to air that I don't find it good in terms of decoupling.

Reply:
You could also just a `util` method that takes a model class and outputs the GQL representation, and store that somewhere in `graphql/`?

Reply:
I would prefer that, yes :) 
I thought you wanted this to be a method on the model classes

Reply:
Addressed in 0ac22a403c8feeef6030535b6bc739f395e35dba

Reply:
> I thought you wanted this to be a method on the model classes

I just mentioned it because that's how we did it previously, but I don't feel strongly about it. Happy for them to be dedicated methods defined in `graphql/`